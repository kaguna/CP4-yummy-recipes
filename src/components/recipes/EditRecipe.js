import React, { Component } from 'react';
import axiosInstance from '../../AxiosInstance'
class EditRecipe extends Component {
    constructor(props){
        super(props);
        this.state = {recipe_name:"", error: ""};
    }
    inputHandler = (event) => {
        event.preventDefault();
        let {name, value} = event.target;
        this.setState({[name]: value});
    };

    editRecipeHandler = (event) => {
        event.preventDefault();
        const {recipe_name} = this.state;
        const header = {headers:{'x-access-token': window.localStorage.getItem('token')},
            content_type: 'application/json'};
        axiosInstance.put("/category/"+this.props.recipe.category_id+"/recipe/"+this.props.recipe.id ,
         {recipe_name})
            .then(response => {
                this.props.parent.setState({mess: response.data.message, error: ""});
                this.props.recipeAfterEdit();
               
                
            })
            .catch(error => {
                if (error.response) {
                    this.setState({error: error.response.data.message, mess: ""});
                    console.log(error.response.data.message)
                } else if (error.request) {
                    this.setState({error:"Can't connect to the server.Please check your connection and try again.", mess: ""});
                }
            });
    };
    render() {
        return (
            
            <div className="modal fade" id={this.props.recipe_id} tabindex = "-1" role="dialog"
                 aria-labelledby="exampleModalCenterTitle" aria-hidden="true">
                <div className="modal-dialog" role="document">
                    <div className="modal-content">
                        <div class="modal-header">
                            <h5 className="modal-title" id="exampleModalLongTitle">Edit category ({this.props.recipe.recipe_name})</h5>
                        </div>
                        <div className="modal-body">
                            {this.state.error?
                                <div className="alert alert-danger">{this.state.error}</div>: ""}
                            
                                <div className="form-group">
                                    <label className="control-label col-sm-3">New Recipe Name:</label>
                                    <div className="col-sm-8">
                                        <input type="text" name="recipe_name" className="form-control"
                                               placeholder="Enter New Recipe name"
                                               defaultValue={this.props.recipe.recipe_name} onChange={this.inputHandler}/>
                                    </div>
                                </div>
                                </div>
                                <div className="modal-footer">
                                    <button className="btn btn-danger" data-dismiss="modal">Close</button>
                                    <button  className="btn btn-primary" data-dismiss="modal" onClick={this.editRecipeHandler}>
                                        <i class="glyphicon glyphicon-edit"></i> Edit Recipe
                                    </button>
                                </div>
                            
                    </div>
                </div>
            </div>
        );
    }
}
export default EditRecipe;
