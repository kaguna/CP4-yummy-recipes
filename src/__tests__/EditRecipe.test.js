import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import EditRecipe from '../components/recipes/EditRecipe';

describe('Edit recipes component', () => {
  const props = {
    recipe: { category_id: 1,
      recipe_description: 'fghbjnk',
      recipe_name: 'cool stuff' },
  };
  const wrapper = shallow(<EditRecipe {...props} />);

  it('renders properly', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
