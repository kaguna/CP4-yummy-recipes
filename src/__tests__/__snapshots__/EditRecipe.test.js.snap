// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit recipes component renders properly 1`] = `
<div
  aria-hidden="true"
  aria-labelledby="exampleModalCenterTitle"
  className="modal fade"
  role="dialog"
  tabIndex="-1"
>
  <div
    className="modal-dialog"
    role="document"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h5
          className="modal-title"
          id="exampleModalLongTitle"
        >
          Edit Recipes (
          cool stuff
          )
        </h5>
      </div>
      <div
        className="modal-body"
      >
        <div
          className="row"
        >
          <div
            className="form-group"
          >
            <label
              className="control-label col-sm-3"
            >
              New Recipe Name:
            </label>
            <div
              className="col-sm-8"
            >
              <input
                className="form-control"
                defaultValue="cool stuff"
                name="recipe_name"
                onChange={[Function]}
                placeholder="Enter New Recipe name"
                type="text"
              />
            </div>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="form-group"
          >
            <label
              className="control-label col-sm-3"
            >
              Recipe Procedure:
            </label>
            <div
              className="col-sm-9"
            >
              <textarea
                className="form-control"
                defaultValue="fghbjnk"
                name="recipe_procedure"
                onChange={[Function]}
                placeholder="Enter recipe procedure"
                rows="10"
                type="text"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="modal-footer"
      >
        <button
          className="btn btn-danger"
          data-dismiss="modal"
        >
          Close
        </button>
        <button
          className="btn btn-primary"
          data-dismiss="modal"
          onClick={[Function]}
        >
          <i
            className="glyphicon glyphicon-edit"
          />
           Edit Recipe
        </button>
      </div>
    </div>
  </div>
</div>
`;
